homeassistant/components/bloomsky.py

```diff
@@ -0,0 +1,75 @@
+"""
+homeassistant.components.bloomsky
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for BloomSky weather station.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/bloomsky/
+"""
+import logging
+from datetime import timedelta
+import requests
+from homeassistant.util import Throttle
+from homeassistant.helpers import validate_config
+from homeassistant.const import CONF_API_KEY
+
+DOMAIN = "bloomsky"
+BLOOMSKY = None
+
+_LOGGER = logging.getLogger(__name__)
+
+# the BloomSky only updates every 5-8 minutes as per the API spec so there's
+# no point in polling the API more frequently
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+
+# pylint: disable=unused-argument,too-few-public-methods
+def setup(hass, config):
+    """
+    Setup BloomSky component.
+    """
+    if not validate_config(
+            config,
+            {DOMAIN: [CONF_API_KEY]},
+            _LOGGER):
+        return False
+
+    api_key = config[DOMAIN][CONF_API_KEY]
+
+    global BLOOMSKY
+    BLOOMSKY = BloomSky(api_key)
```

Comment:
How about to check if the connection was made successfully? And if not due to a wrong/invalid API key, let the initialization of the component fail.